tests/Microsoft.Identity.Test.Unit.net45/AppConfigTests/ConfidentialClientApplicationBuilderTests.cs

```diff
@@ -0,0 +1,195 @@
+// 
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+// 
+// This code is licensed under the MIT License.
+// 
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+// 
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+// 
+// ------------------------------------------------------------------------------
+
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Identity.Client;
+using Microsoft.Identity.Client.AppConfig;
+using Microsoft.Identity.Client.Core;
+using Microsoft.VisualStudio.TestTools.UnitTesting;
+
+namespace Microsoft.Identity.Test.Unit.AppConfigTests
+{
+    [TestClass]
+    public class ConfidentialClientApplicationBuilderTests
+    {
+        [TestMethod]
+        public void TestConstructor()
+        {
+            var cca = ConfidentialClientApplicationBuilder.Create(MsalTestConstants.ClientId).Build();
+            Assert.AreEqual(MsalTestConstants.ClientId, cca.ClientId);
+            Assert.IsNotNull(cca.UserTokenCache);
+
+            // Validate Defaults
+            Assert.AreEqual(LogLevel.Warning, cca.AppConfig.LogLevel);
+            Assert.IsNull(cca.AppConfig.ClientCredential);
+            Assert.AreEqual(MsalTestConstants.ClientId, cca.AppConfig.ClientId);
+            Assert.IsNull(cca.AppConfig.Component);
+            Assert.AreEqual(false, cca.AppConfig.EnablePiiLogging);
+            Assert.IsNull(cca.AppConfig.HttpClientFactory);
+            Assert.AreEqual(false, cca.AppConfig.IsDefaultPlatformLoggingEnabled);
+            Assert.IsNull(cca.AppConfig.LoggingCallback);
+            Assert.AreEqual(Constants.DefaultRedirectUri, cca.AppConfig.RedirectUri);
+            Assert.IsNull(cca.AppConfig.TelemetryCallback);
+            Assert.AreEqual(null, cca.AppConfig.TenantId);
+        }
+
+        [TestMethod]
+        public void TestWithDifferentClientId()
+        {
+            var cca = ConfidentialClientApplicationBuilder.Create("this is a test client id").Build();
```

Comment:
>this is a test client id [](start = 67, length = 24)

all these can move to a const file at some point, or use ones we have in other tests?

Reply:
i wanted it to be some rando string here to ensure we're not defaulting it to somethign (like msaltestconstants.clientid)

---
In reply to: [249203335](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/800#discussion_r249203335) [](ancestors = 249203335)

Reply:
nit: local method const makes it easier to re-use with out the risk of a typo.

---
In reply to: [249210045](https://github.com/AzureAD/microsoft-authentication-library-for-dotnet/pull/800#discussion_r249210045) [](ancestors = 249210045,249203335)